fuzz/fuzz_targets/execute.rs

```diff
@@ -1,57 +1,68 @@
-#![no_main]
-
 mod utils;
 
-use libfuzzer_sys::fuzz_target;
-use utils::{arbitrary_exec_module, ty_to_val};
-use wasmi::{Engine, Linker, Module, Store, StoreLimitsBuilder};
+use arbitrary::Unstructured;
+use honggfuzz::fuzz;
+use utils::{arbitrary_swarm_config_module, ty_to_arbitrary_val};
+use wasmi::{Config, Engine, Linker, Module, Store, StoreLimitsBuilder};
 
-fuzz_target!(|data: &[u8]| {
-    let Ok(mut smith_module) = arbitrary_exec_module(data) else {
-        return;
-    };
+fn main() {
+    loop {
+        fuzz!(|seed: &[u8]| {
+            let mut unstructured = Unstructured::new(seed);
+            let Ok(smith_module) = arbitrary_swarm_config_module(&mut unstructured) else {
+                return;
+            };
+            let wasm = smith_module.to_bytes();
 
-    // TODO: We could use Wasmi's built-in fuel metering instead.
-    //       This would improve test coverage and may be more efficient
-    //       given that `wasm-smith`'s fuel metering uses global variables
-    //       to communicate used fuel.
-    let Ok(_) = smith_module.ensure_termination(1000 /* fuel */) else {
-        return;
-    };
-    let wasm = smith_module.to_bytes();
-    let engine = Engine::default();
-    let linker = Linker::new(&engine);
-    let limiter = StoreLimitsBuilder::new()
-        .memory_size(1000 * 0x10000)
-        .build();
-    let mut store = Store::new(&engine, limiter);
-    store.limiter(|lim| lim);
-    let module = Module::new(store.engine(), wasm.as_slice()).unwrap();
-    let Ok(preinstance) = linker.instantiate(&mut store, &module) else {
-        return;
-    };
-    let Ok(instance) = preinstance.ensure_no_start(&mut store) else {
-        return;
-    };
+            let mut config = Config::default();
+            config.consume_fuel(true);
```

Comment:
With the above `smith_module.ensure_termination(1000 /* fuel */)` we now have a double fuel-based protection. Not sure if worth it. 🤔

Reply:
Good catch - just a copy-and-paste error from moving changes to this branch.

Reply:
Actually, on second glance, not sure which part you're referring to. In this harness, the `smith_module.ensure_termination` call has been removed and replaced with the built-in fuel metering.

Reply:
You are right, for the `execution` fuzz job that should be fine and maybe even better because the fuzzer covers more code this way. For differential fuzzing it would be bad.